src/ingestors/fxhash/index.ts

```diff
@@ -0,0 +1,147 @@
+import { MintContractOptions, MintIngestor, MintIngestorResources } from '../../lib/types/mint-ingestor';
+import { MintIngestionErrorName, MintIngestorError } from '../../lib/types/mint-ingestor-error';
+import { MintInstructionType, MintTemplate } from '../../lib/types/mint-template';
+import { MintTemplateBuilder } from '../../lib/builder/mint-template-builder';
+import { FXHASH_BASE_FIXED_PRICE_ABI, FXHASH_BASE_FRAME_ABI } from './abi';
+import { getFxHashMintPriceInEth } from './onchain-metadata';
+
+const BASE_FRAME_CONTRACT_ADDRESS = '0x6e625892C739bFD960671Db5544E260757480725';
+const BASE_FIXED_PRICE_CONTRACT_ADDRESS = '0x4bDcaC532143d8d35ed759189EE22E3704580b9D';
+
+export class FxHashIngestor implements MintIngestor {
+  async supportsUrl(_resources: MintIngestorResources, url: string): Promise<boolean> {
+    return new URL(url).hostname === 'www.fxhash.xyz' || new URL(url).hostname === 'fxhash.xyz';
+  }
+
+  async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+    if (contract.chainId !== 8453) {
```

Comment:
`supportsContract` should _either_:

* Determine whether the contract is an FXHash contract (by searching the graph) <- this is ideal
* OR... always return false to disable ingestion by contract

In this case, you likely want to look up the contract address in the graph, and then in `createMintForContract`, get the URL from the graph and then just make an internal call to `createMintForUrl`?

Not the most economical for HTTP calls, but simplest for code here?

This will allow ingestion of mints based on blockchain indexing (so automatically) vs. users manually submitting mints as URLs

Reply:
Yeah - I just wasn't sure on your ingestion. So the `contract.contractAddress` here would be the token's address right?

<img width="545" alt="Screenshot 2024-06-24 at 10 57 14 AM" src="https://github.com/floornfts/mobile-minting/assets/171729394/162cb0e0-5c7f-415e-8595-90a7c49951bd">

Reply:
Yeah exactly! 🙏 

Infact, that's a great callout. We probably need to clarify the idea of `mintInstructions.contractAddress` and the collection's `tokenAddress` 

BUT... for now, will leave as is so that you can hit a static target, and then when we change that, I'll update all ingestors 🙏 

In this case, yes `contract.contractAddress` input will be the minted collection's contract address 🥳

Reply:
Done!